desc/protoparse/registered_descriptor.go

```diff
@@ -0,0 +1,21 @@
+package protoparse
+
+import (
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/internal"
+)
+
+// GetRegisteredDescriptors takes a list of proto filenames which have
+// already been imported as compiled Go protobufs, and returns a map
+// of FileDescriptorProto objects.
+func GetRegisteredDescriptors(filenames ...string) (map[string]*dpb.FileDescriptorProto, error) {
```

Comment:
Please remove this method: (1) It doesn't belong in `protoparse` since it doesn't have anything to do with parsing proto sources; (2) it is very redundant with the `desc` package, which already provides methods for loading file descriptors by name.

Reply:
Is there another place it could go? It does seem really useful to have, since otherwise folks need to copy the contents of the internal methods in their own code in order to do this.

Reply:
I also might be missing something: I can not figure out which method in `desc` this is redundant with. I see a method for getting a file descriptor from the `FileDescriptorProto`, but this is getting a `FileDescriptorProto` from a registered proto's filename. As best as I can tell, `desc` does not currently expose this.

Perhaps it would be reasonable to have the `desc` package include a method to get a `*desc.FileDescriptor` from a registered filename? (If I do that, I would also change the parse method to accept that instead of a `*dpb.FileDescriptorProto`.

Reply:
https://godoc.org/github.com/jhump/protoreflect/desc#LoadFileDescriptor

Reply:
Acknowledged, will incorporate.

Reply:
Done.